src/dmd/backend/blockopt.c

```diff
@@ -727,7 +727,7 @@ void brcombine()
                     {
                         if (EOP(b3->Belem))
                             continue;
-                        tym_t ty = (bc2 == BCretexp) ? b2->Belem->Ety : TYvoid;
+                        tym_t ty = (bc2 == BCretexp) ? b2->Belem->Ety : static_cast<regm_t>(TYvoid);
```

Comment:
```
dmd/backend/blockopt.c: In function ‘void brcombine()’:
dmd/backend/blockopt.c:730:54: error: enumeral and non-enumeral type in conditional expression [-Werror=extra]
                         tym_t ty = (bc2 == BCretexp) ? b2->Belem->Ety : TYvoid;
                                    ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

`-Wno-enum-compare` didn't have any effect.

Reply:
> -Wno-enum-compare didn't have any effect.

That's because it seems to be turned on by `-Wextra`.   Or at least according to this part of the error `[-Werror=extra]`.